components/automate-ui/src/app/components/pending-edits-bar/pending-edits-bar.component.ts

```diff
@@ -0,0 +1,67 @@
+import { Component, OnInit, OnDestroy, ViewEncapsulation } from '@angular/core';
+import { interval as observableInterval, Subject } from 'rxjs';
+import { take, takeUntil } from 'rxjs/operators';
+import { Store } from '@ngrx/store';
+import { NgrxStateAtom } from 'app/ngrx.reducers';
+
+import { ProjectService } from 'app/entities/projects/project.service';
+import { Project } from 'app/entities/projects/project.model';
+import { allProjects } from 'app/entities/projects/project.selectors';
+
+@Component({
+  selector: 'app-pending-edits-bar',
+  templateUrl: './pending-edits-bar.component.html',
+  styleUrls: ['./pending-edits-bar.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class PendingEditsBarComponent implements OnInit, OnDestroy {
+  public showPendingEditsBar = false;
+  public confirmApplyStartModalVisible = false;
+  private isDestroyed = new Subject<boolean>();
+
+  constructor(
+    private store: Store<NgrxStateAtom>,
+    public projects: ProjectService
+    ) {
+      this.store.select(allProjects).pipe(
+        takeUntil(this.isDestroyed)
+      ).subscribe((projectList: Project[]) => {
+        this.showPendingEditsBar = projectList.some(p => p.status === 'EDITS_PENDING');
+      });
+    }
+
+  ngOnInit(): void {}
+
+  ngOnDestroy(): void {
+    this.isDestroyed.next(true);
+    this.isDestroyed.complete();
+  }
+
+  public openConfirmUpdateStartModal(): void {
+    this.confirmApplyStartModalVisible = true;
+  }
+
+  private closeConfirmApplyStartModal(): void {
+    this.confirmApplyStartModalVisible = false;
+  }
+
+  public confirmApplyStart(): void {
+    this.closeConfirmApplyStartModal();
+    this.projects.applyRulesStart();
+
+    // Rapid sampling for 3 seconds for more responsive UX.
+    // If the update is still running, the secondary (active) emitter
+    // will check this status at frequent intervals.
+    // Once the update completes, the tertiary (dormant) emitter
+    // will check this status at INfrequent intervals.
+    // (See getActiveApplyRulesStatus$ and getDormantApplyRulesStatus$.)
+    observableInterval(250).pipe(take(12)) // 12 x 250ms => 3 seconds
+      .subscribe(() => {
+        this.projects.getApplyRulesStatus();
+      });
```

Comment:
Neat!

Reply:
After this 3-second checking, polling of project status moves to project.effects.ts. There, both the active and dormant emitters poll the same thing. The only difference is the rate: while projects are being updated (i.e. an update is active), the status is checked every 5 seconds (so it can more responsively show changes during the update). When projects are not being updated, polling slows down to once a minute to reduce overhead. That cycle repeats the next time someone (not necessarily the current user) starts another project update.